### ISSUE TITLE
Asp.Net Core 8 app crashes after upgrading package from 3.3.2 to 4.0.0 (and 4.0.3)

### ISSUE THREAD

**Issue by olejhar** [chriswill/serilog-sinks-azureblobstorage#126]
I'm hosting a .Net Core 8 Web Api in an Azure app service. The app is configured to use Serilog to write to an Azure blob, and with version 3.3.2 of this package everything has worked fine. I recently tried to upgrade the nuget packages for the project, and skipped directly to 4.0.3. This crashed my application, and after looking into the logs, I managed to isolate the issue to something related to a missing file, with the exception:
```
System.IO.FileNotFoundException: Could not load file or assembly 'Serilog.Sinks.PeriodicBatching, Version=4.1.1.0, Culture=neutral, PublicKeyToken=24c2f752a8e58a10'. The system cannot find the file specified.
```

The exception stack looks like this:
```
System.IO.FileNotFoundException: Could not load file or assembly 'Serilog.Sinks.PeriodicBatching, Version=4.1.1.0, Culture=neutral, PublicKeyToken=24c2f752a8e58a10'. The system cannot find the file specified.
File name: 'Serilog.Sinks.PeriodicBatching, Version=4.1.1.0, Culture=neutral, PublicKeyToken=24c2f752a8e58a10'
   at System.Reflection.RuntimeAssembly.InternalLoad(AssemblyName assemblyName, StackCrawlMark& stackMark, AssemblyLoadContext assemblyLoadContext, RuntimeAssembly requestingAssembly, Boolean throwOnFileNotFound)
   at System.Reflection.Assembly.Load(AssemblyName assemblyRef)
   at Serilog.Settings.Configuration.ConfigurationReader.LoadConfigurationAssemblies(IConfiguration section, AssemblyFinder assemblyFinder)
   at Serilog.Settings.Configuration.ConfigurationReader..ctor(IConfiguration configSection, AssemblyFinder assemblyFinder, ConfigurationReaderOptions readerOptions, IConfiguration configuration)
   at Serilog.ConfigurationLoggerConfigurationExtensions.GetConfigurationReader(IConfiguration configuration, ConfigurationReaderOptions readerOptions, DependencyContext dependencyContext)
   at Serilog.ConfigurationLoggerConfigurationExtensions.Configuration(LoggerSettingsConfiguration settingConfiguration, IConfiguration configuration, ConfigurationReaderOptions readerOptions)
   at Program.<>c__DisplayClass1_0.<AddServices>b__0(IServiceProvider services, LoggerConfiguration lc) in (App specific path removed)\Program.cs:line 67
```
This led me to downgrade this package one version at a time, and it was not until I had reverted back to using 3.3.2 that the application finally started up again.

It looks like this happens during configuration of Serilog, where I read from an environment-specific appsettings file. 

I can use Serilog.AspNetCore version 8.0.3 with Serilog.Sinks.Azureblobstorage 3.3.2 just fine, so I _think_ the issue is related to this particular sink, and something that changed between 3.3.2 and 4.0.0, not Serilog or its configuration in general. 

Looking into the release notes for version 4.0.0 I read that from this release the package has "implemented support for Serilog native IBatchedLogEventSink", and that "This major update may require you to alter your configuration settings where you define usage of this sink.".

I've looked into the examples, as far as I've managed, and have concluded with the following:

- I'm using Managed Identity, so the correct parameter to use in the appsettings file is `storageAccountUri`, not `connectionString`
- Parameter `writeInBatches` is now deprecated
- Parameters `period` and `batchPostingLimit` are now optional.

This is my configuration:
```
"Serilog": {
  "Using": [ "Serilog.Sinks.AzureBlobStorage" ],
  "MinimumLevel": {
    "Default": "Information",
    "Override": {
      "Microsoft.AspNetCore": "Warning",
      "Microsoft.EntityFrameworkCore": "Warning",
      "Microsoft.IdentityModel": "Warning"
    }
  },
  "WriteTo": [
    {
      "Name": "AzureBlobStorage",
      "Args": {
        "formatter": "Serilog.Formatting.Json.JsonFormatter",
        "storageAccountUri": "https://xxxxxxxxx.blob.core.windows.net/",
        "storageContainerName": "api-logs",
        "storageFileName": ""
        //"writeInBatches": "true",
        //"period": "0.00:00:30",
        //"batchPostingLimit": "50"
      }
    }
  ],
  "Enrich": [ "FromLogContext", "WithMachineName" ],
  "Properties": {
    "ApplicationName": "XXXXXXXX"
  }
}
```
For the Serilog initalization, I do use the two-stage initialization procedure. First I create a bootstrapped logger right after creating the web application builder:
```
var builder = WebApplication.CreateBuilder(args);

// Add serilog
Log.Logger = new LoggerConfiguration().WriteTo.Console().CreateBootstrapLogger();
```
and replace this with a proper logger:
```
builder.Services.AddSerilog((services, lc) => lc
.ReadFrom.Configuration(builder.Configuration)
.WriteTo.Console());  
```
I'm wondering if I need to use `.WriteTo.AzureBlobStorage()` here? With the managedIdentityClientId parameter?

Is there anything that stands out here, or any suggestions? 

I am fine with using version 3.3.2 for the time being, but would like to raise this as an issue.

Thanks!

**Comment by chriswill** [comment]
Your overall appsettings.json configuration looks fine, as does what you are doing in Program.cs.  There is no need to add .WriteTo.AzureBlobStorage() in Program.cs if you have it properly defined in appsettings.json and are reading from configuration.

There is a small sample in the source code that you might want to play with: https://github.com/chriswill/serilog-sinks-azureblobstorage/tree/main/samples/SampleConsoleApp

It initializes Serilog as you do, and there is a commented out section in appsettings.json that you can enable to test managed identity.  https://github.com/chriswill/serilog-sinks-azureblobstorage/blob/main/samples/SampleConsoleApp/appsettings.json

If you are using a managed identity client id, you can specify that in appsettings.json using the "managedIdentityClientId" property.  Otherwise, you'll be using a DefaultAzureCredential, which may not be a bad thing.  There was a prior discussion about that here: https://github.com/chriswill/serilog-sinks-azureblobstorage/issues/121

I've just verified that the sample above works in Visual Studio using DefaultAzureCredential without throwing that System.IO.FileNotFoundException exception you reference in your post.  The current version of the Azure Blob sink doesn't have any references to Serilog.Sinks.PeriodicBatching any longer, so I'm wondering if somewhere in your app you are getting a reference to an older version of Serilog.Settings.Configuration or some other Serilog library.

Use the sample app to verify that your env is working as expected, and then replicate that to your Web API project.

**Comment by chriswill** [comment]
Also, I just released a new version 4.0.5, which has newer Nugets to remove a security vulnerability in System.Text.Json.
